cmd/gcp-controller-manager/app/gke_approver.go

```diff
@@ -673,6 +673,43 @@ func ensureNodeMatchesMetadataOrDelete(opts GCPConfig, csr *capi.CertificateSign
 }
 
 func shouldDeleteNode(opts GCPConfig, node *v1.Node) (bool, error) {
-	// TODO: return true if the Node API object does not match metadata and should be deleted
+	inst, err := getInstanceByName(opts, node.Name)
+	if err != nil {
+		glog.Errorf("Error retrieving instance %q: %v. Falling back to not deleting the node: %v", node.Name, err)
+		return false, nil
+	}
+	aliasRanges := getAliasRangesFromInstance(inst)
+	if len(aliasRanges) != 1 {
+		glog.Errorf("Got %d alias IP ranges from instance %q, expecting 1. Falling back to not deleting the node.", len(aliasRanges), inst.Name)
+	}
+	if aliasRanges[0] != node.Spec.PodCIDR {
```

Comment:
How about combining this with the loop in getAliasRangesFromInstance ?

```
	for _, networkInterface := range inst.NetworkInterfaces {
		for _, r := range networkInterface.AliasIpRanges {
			if node.Spec.PodCIDR == r.IpCidrRange {
                           return false, nil
                        }
		}
	}
        return true, nil
```

Reply:
Sure, I just realize I haven't considered new node here --- podCIDR might be empty initially. I will update the logic.

Reply:
Updated commit, PTAL thanks!